index.js

```diff
@@ -247,6 +257,30 @@ class LedgerBridgeKeyring extends EventEmitter {
             }
           })
         })
+        .catch(reject)
+    })
+  }
+
+  unlockAccountByAddress (address) {
+    return new Promise((resolve, reject) => {
+
+      const checksummedAddress = ethUtil.toChecksumAddress(address)
+      if (!Object.keys(this.accountDetails).includes(checksummedAddress)) {
+        reject(new Error(`Ledger: Account for address '${checksummedAddress}' not found`))
+        return
+      }
+      const { hdPath } = this.accountDetails[checksummedAddress]
+
+      this.unlock(hdPath)
+        .then((unlockedAddress) => {
+          // unlock resolves to the address for the given hdPath as reported by the ledger device
+          // if that address is not the requested address, then this account belongs to a different device or seed
+          if (unlockedAddress.toLowerCase() !== address.toLowerCase()) {
+            reject(new Error(`Ledger: Account ${address} does not belong to the connected device`))
```

Comment:
It looks like there is a `return` missing after this line.

Note that one way to prevent this would be to convert this function to use `async/await` rather than a Promise constructor. Generally `async/await` would be preferred in call cases but I wouldn't block on it. As you can see, we use both patterns interchangeably at the moment.

Reply:
Thanks for your feedback! I changed the function to use async/await as you suggested.